src/github.com/getlantern/errlog/errlog.go

```diff
@@ -0,0 +1,529 @@
+/*
+Package errlog defines error types used across Lantern project and implements
+functions to manipulate them.
+
+  var elog = errlog.ErrorLoggerFor("package-name")
+  if n, err := Foo(); err != nil {
+    elog.Log(err, errlog.WithOp("foo"), errlog.WithField("foo": "bar"))
+  }
+
```

Comment:
@oxtoacart @uaalto How do you think about above idiom to report errors? It follows the usage of `golog`. Besides reporting to globally supplied reporter, it calls `golog.Error()` internally, so we can completely replace `log.Errorf` with it. More detailed usage is in `errlog_test.go`. I also replaced `log.Errorf` of a few packages as examples.